# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/fetch/FetchDriver.h
# Commit: 34d4e99f8219
# Full Hash: 34d4e99f8219f7fb87e59e16f8a9ab32f823c538
# Author: Eden Chuang <echuang@mozilla.com>
# Date: 2022-02-26 21:36:36
# Regressor Bug: 1754365
# File Overlap Count: 1
# Description:
#   Bug 1754365 - Getting PerformanceTimingData from channel for FetchService. r=dom-worker-reviewers,jesup
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D138249
# ==============================================================================

diff -r a714e906b18f -r 34d4e99f8219 dom/fetch/FetchDriver.h
--- a/dom/fetch/FetchDriver.h	Sat Feb 26 07:22:44 2022 +0000
+++ b/dom/fetch/FetchDriver.h	Sat Feb 26 08:40:25 2022 +0000
@@ -38,6 +38,7 @@
 class InternalRequest;
 class InternalResponse;
 class PerformanceStorage;
+class PerformanceTimingData;
 
 /**
  * Provides callbacks to be called when response is available or on error.
@@ -124,6 +125,9 @@
     mOriginStack = std::move(aOriginStack);
   }
 
+  PerformanceTimingData* GetPerformanceTimingData(nsAString& aInitiatorType,
+                                                  nsAString& aEntryName);
+
   // AbortFollower
   void RunAbortAlgorithm() override;
 